docs/installation.rst

```diff
@@ -1,11 +1,46 @@
 Installation
 ============
 
-To install this package run
+To use StarkNet.py, ``ecdsa, fastecdsa, sympy`` dependencies are required. Depending on the operating system,
+different install steps must be performed.
 
-``pip install starknet.py``
+Linux
+^^^^^
 
-or using Poetry:
+.. code-block:: console
 
-``poetry add starknet.py``
+    sudo apt install -y libgmp3-dev
+    pip install starknet-py
 
+MacOS with Intel processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+
+.. code-block:: console
+
+    brew install gmp
+    pip install starknet-py
+
+MacOS with Apple silicon
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+First install gmp:
+
+.. code-block:: console
+
+    brew install gmp
```

Comment:
I guess this could be resrtructured to have:

```
# Macos:
<install gmp>
## Apple silicon
<additional steps>
```
As the step with gmp is identical regardless of cpu.

Reply:
I though about this, but I am unsure. I wanted to have the complete instruction for every platform in each section.

Reply:
I would do 
```
#MacOS
## Apple silicon
## x86
```

Reply:
I'll do something along the lines of this if you both prefer having subsections.